### PR TITLE
Add watchdog as a bundled dependency on Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds watchdog to the requirements file for Windows. This will add some `iNotify` behavior to Windows via the watchdog beacon.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/52804

### Tests written?
NA

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by dwoz — APPROVED** [review]

**Comment by twangboy** [comment]
re-run all

**Comment by twangboy** [comment]
re-run full all

**Comment by Ch3LL** [comment]
re-run full all

**Comment by Ch3LL** [comment]
i want to make sure these tests run `tests/unit/beacons/test_watchdog.py` and do not get skipped.

**Comment by twangboy** [comment]
@Ch3LL This should get the tests runnin: https://github.com/saltstack/salt-jenkins/pull/1404

File: tests/unit/beacons/test_watchdog.py [hunk 1]
```diff
@@ -162,28 +178,49 @@ def test_trigger_all_possible_events(self):
...
-        self.assertEqual(ret[5]['change'], 'modified')
-
-        # delete
-        self.assertEqual(ret[6]['path'], moved)
-        self.assertEqual(ret[6]['change'], 'deleted')
-        self.assertEqual(ret[7]['path'], self.tmpdir)
-        self.assertEqual(ret[7]['change'], 'modified')
+        # Windows only returns a single item for each event
+        # Other OS'es return 2 apparently
+        if salt.utils.platform.is_windows():
```
Comment (s0undt3ch) [inline tests/unit/beacons/test_watchdog.py]: I would create two separate test cases...

**Review by s0undt3ch — APPROVED** [review]

File: tests/unit/beacons/test_watchdog.py (hunk 1 above)
Comment (twangboy) [reply tests/unit/beacons/test_watchdog.py]: Done

File: requirements/static/windows.in [hunk 2]
```diff
@@ -23,3 +23,12 @@ setproctitle
 strict_rfc3339
 supervisor==3.3.5; python_version < '3'
 virtualenv
+watchdog
+
+# If running under windows, please uncomment the following 2 requirements before running
```
Comment (s0undt3ch) [inline requirements/static/windows.in]: You don't really need to do this anymore.
In fact, will `watchdog` be a windows packaging dependency?
If so, let's move it to `pkg/windows/reqs.txt` and recompile requirements.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: requirements/static/windows.in (hunk 2 above)
Comment (s0undt3ch) [reply requirements/static/windows.in]: The comment at the end?

**Review by s0undt3ch — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Sorry, I didn't think this through.
We need to keep `watchdog` out of `windows.in` but it should be included in the linux'es `.in` files and also OSX `.in` files if it's not present in `pkg/osx/reqs.txt`

So sorry.

**Review by Akm0d — APPROVED** [review]

**Pull request merged**